Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -209,8 +210,60 @@ CFArrayRef _DWriteCopyFontNamesForFamilyName(CFStringRef familyName) {
  * Note: This function currently uses a cache, meaning that fonts installed during runtime will not be reflected,
  * unless registered to the user font collection
  */
-CFStringRef _DWriteGetFamilyNameForFontName(CFStringRef fontName) {
-    return _DWriteGetFontPropertiesFromName(fontName)->familyName.get();
+CFStringRef _DWriteGetCompatibleFamilyName(CFStringRef fontName, IDWriteFontFace* fontFace) {
+    woc::unique_cf<CFStringRef> upperFontName(_CFStringCreateUppercaseCopy(fontName));
+
+    const auto& info = __GetSystemFontCollectionHelper()->GetFontPropertiesFromUppercaseFontName(upperFontName);
+    if (info) {
+        return info->familyName.get();
+    }
+
+    const auto& userFontInfo = __GetUserFontCollectionHelper()->GetFontPropertiesFromUppercaseFontName(upperFontName);
+    if (userFontInfo) {
+        if (!fontFace) {
+            return userFontInfo->familyName.get();
+        }
+
+        // Retrieve the font file that was used to create this font face
+        UINT32 fileCount;
+        fontFace->GetFiles(&fileCount, nullptr);
+        RETURN_RESULT_IF(fileCount == 0, userFontInfo->familyName.get());
```

Comment:
mix of macros and ifs deep within the function.
why cant above be the same or vise versa?